api/src/services/lottery.service.ts

```diff
@@ -584,4 +792,144 @@ export class LotteryService {
 
     return res;
   }
+
+  getActivityLogKey = (logData: Prisma.JsonValue | null) => {
+    if (!logData) return null;
+    return logData[Object.keys(logData)[0]];
+  };
+
+  getLotteryStatusFromActivityLogMetadata = (
+    status: LotteryActivityLogStatus,
+    index: number,
+    previousStatus: LotteryActivityLogStatus | null,
+  ): LotteryActivityLogStatus => {
+    if (!status) return;
+    if (index === 0) return status;
+
+    if (
+      status === LotteryStatusEnum.ran &&
+      (previousStatus === LotteryStatusEnum.releasedToPartners ||
+        previousStatus === LotteryStatusEnum.publishedToPublic)
+    ) {
+      return 'retracted';
+    }
+    if (
+      status === LotteryStatusEnum.ran &&
+      previousStatus === LotteryStatusEnum.ran
+    ) {
+      return 'rerun';
+    }
+    return status;
+  };
+
+  /*
+   * @param listingId
+   * @returns a list of activity log entries and the name of the user who did the action for lotteries
+   */
+  public async lotteryActivityLog(
+    listingId: string,
+    requestingUser: User,
+  ): Promise<LotteryActivityLogItem[]> {
+    const activityLogs = await this.prisma.activityLog.findMany({
+      select: {
+        metadata: true,
+        updatedAt: true,
+        userAccounts: {
+          select: {
+            firstName: true,
+            lastName: true,
+          },
+        },
+      },
+      where: {
+        AND: {
+          OR: [{ module: 'listing', action: 'update' }, { module: 'lottery' }],
+          AND: [{ recordId: listingId }, { recordId: { not: null } }],
+        },
+      },
+      orderBy: {
+        updatedAt: OrderByEnum.ASC,
+      },
+    });
+
+    const filteredActivityLogs = activityLogs
+      .sort((a, b) => a.updatedAt.getTime() - b.updatedAt.getTime())
+      .filter((log) => {
+        const logString = JSON.stringify(log.metadata);
+        // only return closed listing status updates
+        if (logString.includes('status')) {
+          if (logString.includes(ListingsStatusEnum.closed)) {
+            return true;
+          } else return false;
+        }
```

Comment:
nit: since this is a json response you don't need to convert it to a string. So you could do something like
```
if (log.metadata['status']) {
  return log.metadata['status'] === ListingsStatusEnum.closed;
}
````